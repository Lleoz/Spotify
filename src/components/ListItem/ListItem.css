.song {
  border-top: 1px solid;
  margin: 10px 0px;
  padding: 10px;
}

.song:hover {
  background-color: rgba(255, 255, 255, 0.1);
  border: 1px solid transparent;
  border-radius: 4px;
}

.details {
  vertical-align: middle;
}

.audio {
  vertical-align: middle;
  margin-left: auto;
  margin-right: 0;
}

.artist {
  display: block;
  font-size: 14px;
}

audio {
  float: right;
}

.name {
  color: #fff;
}
