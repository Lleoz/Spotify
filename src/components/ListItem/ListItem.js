import React from 'react'
import { useDispatch } from 'react-redux'
import { modalOpen } from '../../actions/modal'
import PropTypes from 'prop-types'


import './ListItem.css'

export const ListItem = ( {id, name, preview_url, artists} ) => {

    const dispatch = useDispatch()

    const openModal = () => {
        dispatch(modalOpen({
            id,
            name,
            preview_url,
            artists
        }))
    }

    return (
        <>
            <div key={id} className='row song'>
                <div className='details col-lg-6'>
                    <span onClick={ openModal } className='name'>{name}</span>
                    <span className='artist'>{artists}</span>
                </div>
                <div className='audio col-lg-6'>
                    <audio className='col-12 float-end' id={name} src={preview_url} controls='1'>
                    </audio>
                </div>
            </div>
        </>
    )
}

ListItem.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    preview_url: PropTypes.string.isRequired,
    artists: PropTypes.string.isRequired
}