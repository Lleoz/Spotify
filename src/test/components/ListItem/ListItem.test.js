import React from 'react'
import { shallow } from "enzyme"
import { ListItem } from '../../../components/ListItem/ListItem';
import { Provider } from 'react-redux';
import { store } from '../../../store/store'
import { useDispatch } from 'react-redux'

describe('Prueba en <ListItem></ListItem>', () => {


    const track = {
        id: '123',
        name: 'Iron',
        preview_url: 'https://github.com/',
        artists: 'Iron Maiden'
    }

    const wrapper = shallow(
            <Provider store={store}>
                <ListItem {...track}/>
            </Provider>).dive();

    test('Se debe cargar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();

    })

})