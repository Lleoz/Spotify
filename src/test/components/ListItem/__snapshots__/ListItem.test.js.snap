// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <ListItem></ListItem> Se debe cargar el componente correctamente 1`] = `
<ListItem
  artists="Iron Maiden"
  id="123"
  name="Iron"
  preview_url="https://github.com/"
/>
`;
